tests/unittests/Foundation/NSCalendarTests.mm

```diff
@@ -481,15 +481,63 @@
     NSCalendar* calendar = [NSCalendar calendarWithIdentifier:NSCalendarIdentifierGregorian];
     NSDate* expectedDate = [calendar dateWithEra:1 year:2010 month:3 day:31 hour:0 minute:0 second:0 nanosecond:0];
 
-    NSRange range = [calendar rangeOfUnit:NSCalendarUnitWeekOfMonth inUnit:NSCalendarUnitYear forDate:expectedDate];
-    ASSERT_EQ(range.location, 1);
-    ASSERT_EQ(range.length, 5);
-
-    range = [calendar rangeOfUnit:NSCalendarUnitWeekOfYear inUnit:NSCalendarUnitMonth forDate:expectedDate];
+    NSRange range = [calendar rangeOfUnit:NSCalendarUnitWeekOfYear inUnit:NSCalendarUnitMonth forDate:expectedDate];
     ASSERT_EQ(range.location, 10);
     ASSERT_EQ(range.length, 5);
 
     range = [calendar rangeOfUnit:NSCalendarUnitHour inUnit:NSCalendarUnitWeekOfMonth forDate:expectedDate];
     ASSERT_EQ(range.location, 0);
     ASSERT_EQ(range.length, 24);
+
+    range = [calendar rangeOfUnit:NSCalendarUnitWeekOfMonth inUnit:NSCalendarUnitMonth forDate:expectedDate];
+    ASSERT_EQ(range.location, 1);
+    ASSERT_EQ(range.length, 5);
+}
+
+ARM_DISABLED_TEST(NSCalendar, BadMatchingOptions) {
+    @try {
+        NSCalendar* calendar = [NSCalendar calendarWithIdentifier:NSCalendarIdentifierGregorian];
+
+        // January 1st 2010
+        NSDate* constantDate = [calendar dateWithEra:1 year:2010 month:1 day:1 hour:0 minute:0 second:0 nanosecond:0];
+        // January 2nd 2010
+        NSDate* expectedNextDay = [calendar dateWithEra:1 year:2010 month:1 day:2 hour:0 minute:0 second:0 nanosecond:0];
+
+        NSDateComponents* comps = [[NSDateComponents alloc] init];
+        comps.day = 2;
+
+        // No matching options
+        NSDate* date = [calendar nextDateAfterDate:constantDate matchingComponents:comps options:0];
+
+        ASSERT_TRUE_MSG(false, "Should not have gotten here. nextDateAfterDate should throw an INVALID_ARG exception");
+    } @catch (NSException* e) {
+        ASSERT_OBJCEQ(e.name, @"NSInvalidArgumentException");
+    }
+}
+
+OSX_DISABLED_TEST(NSCalendar, NanoSecondComparison) {
+    // Nanosecond comparison is disabled on OSX as the behavior does not match the documentation. Nanosecond comparison will
+    // always return NSOrderedSame if the two dates are within the same second of each other. Thus, even though the two
+    // nanosecond values below are different, OSX will return that this comparison to the granularity of Nanoseconds will
+    // is the same.
+
+    NSCalendar* calendar = [NSCalendar calendarWithIdentifier:NSCalendarIdentifierGregorian];
+    NSDate* date1 = [NSDate dateWithTimeIntervalSinceReferenceDate:.1];
+    NSDate* date2 = [NSDate dateWithTimeIntervalSinceReferenceDate:.9];
+    ASSERT_EQ([calendar compareDate:date1 toDate:date2 toUnitGranularity:NSCalendarUnitNanosecond], NSOrderedAscending);
+    ASSERT_EQ([calendar compareDate:date2 toDate:date1 toUnitGranularity:NSCalendarUnitNanosecond], NSOrderedDescending);
+}
+
+DISABLED_TEST(NSCalendar, rangeOfWeekOfMonthTest) {
```

Comment:
Yes, I'll enable on OSX.

---
In reply to: [74657115](https://github.com/Microsoft/WinObjC/pull/844#discussion_r74657115) [](ancestors = 74657115)